lib/logstash/plugin_mixins/rabbitmq_connection.rb

```diff
@@ -136,7 +152,24 @@ def rabbitmq_settings
       end
 
       def connect!
-        @hare_info = connect() unless @hare_info # Don't duplicate the conn!
+        if !@hare_info # Don't duplicate the conn!
+            if !@connection_pool_name.nil?
+              cache_key = {:host => @host,:port => @port,:vhost => @vhost,:user => @user,:pool => @connection_pool_name}.freeze 
```

Comment:
Would be nice to extract this cache_key to a method since its used in two places. That way if the code is changed there won't be accidental bugs with connect using a different key than close.